examples/neural_sde_vae.ipynb

```diff
@@ -0,0 +1,551 @@
+{
+ "cells": [
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "# Neural SDE (VAE)\n",
+    "\n",
+    "This implementation is based on the Pytorch version of Latent SDE from [`torchsde`](https://github.com/google-research/torchsde/blob/master/examples/latent_sde.py) library. \n",
+    "\n",
+    "The training follows a Bayesian inference procedure where we assume the prior of an SDE has a simple drift function (`drift_prior`).\n",
+    "\n",
+    "The posterior is approximated by a neural SDE with a more complex drift function (`drift_posterior`). Note that both prior and posterior share the same diffusion.\n",
+    "\n",
+    "The objective function is computed via evidence lower bound (ELBO) which requires getting Kullback-Leibler (KL) divergence between the prior and the approximated posterior.\n",
+    "\n",
+    "The KL divergence turns out to be a result of Girsanov's theorem. The function `diffrax.misc.sde_kl_divergence` allows us to augment the original SDE with KL estimation.\n",
+    "\n",
+    "This notebook for this example can be found [here](https://github.com/patrick-kidger/diffrax/blob/main/examples/neural_sde_vae.ipynb). See [this](https://docs.kidger.site/diffrax/examples/neural_sde_gan/) for an alternative way of training neural SDEs using GAN techniques.\n",
+    "\n",
+    "**Reference**\n",
+    "```bibtex\n",
+    "@article{li2020scalable,\n",
+    "  title={Scalable gradients for stochastic differential equations},\n",
+    "  author={Li, Xuechen and Wong, Ting-Kam Leonard and Chen, Ricky T. Q. and Duvenaud, David},\n",
+    "  journal={International Conference on Artificial Intelligence and Statistics},\n",
+    "  year={2020}\n",
+    "}\n",
+    "```"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "import os\n",
+    "import math\n",
+    "\n",
+    "import jax\n",
+    "import optax\n",
+    "import numpy as np\n",
+    "import jax.numpy as jnp\n",
+    "import jax.random as jrandom\n",
+    "import equinox as eqx\n",
+    "\n",
+    "from diffrax import (MultiTerm, ODETerm, ControlTerm,\n",
+    "                     diffeqsolve, Euler,\n",
+    "                     SaveAt, VirtualBrownianTree)\n",
+    "from diffrax.misc import sde_kl_divergence\n",
```

Comment:
So anything not imported as `diffrax.*` is considered private API. If we're going to expose this publicly then `sde_kl_divergence` should be offered as `diffrax.sde_kl_divergence`.